src/components/common/spinner/index.tsx

```diff
@@ -1,5 +1,30 @@
-const Spinner = () => {
-  return <div></div>;
+import { Icon } from '@/components/common';
+import * as variants from '@/styles/variants.css';
+import { CSSProperties } from 'react';
+
+export type SpinnerProps = {
+  size?: 'xSmall' | 'small' | 'medium' | 'large' | 'xLarge' | 'huge';
+  color?: string;
+  loading?: boolean;
+  style?: CSSProperties;
+};
+
+const Spinner = ({
+  size = 'medium',
+  color = variants.vars.color.bg.select,
+  loading = true,
+  ...props
+}: SpinnerProps) => {
+  return loading ? (
+    <Icon
+      className="fas fa-spin"
+      name="circle-notch"
+      size={size}
+      color={color}
+      style={{ color, ...props.style }}
+      {...props}
+    />
+  ) : null;
```

Comment:
p5;
```suggestion
  return loading && (
    <Icon
      className="fas fa-spin"
      name="circle-notch"
      size={size}
      color={color}
      style={{ color, ...props.style }}
      {...props}
    />
  )
```
로도 가능할 것 같습니다😄

Reply:
이렇게도 해봤는데 type이 boolean 형태라고 떠서 고쳤어요 ㅠ